### PR TITLE
Add support for origin Authfile generation script.

### PR DESCRIPTION
**Pull request by bbockelm**
This is the origin-side support for https://github.com/opensciencegrid/topology/pull/335.  With this, the origin server can generate its Authfile solely from topology data.

### CODE REVIEW

File: configs/config.d/50-stash-origin-authz.cfg [hunk 1]
```diff
@@ -32,12 +32,22 @@ if named stash-origin-auth
 
    sec.protbind * gsi
 
-   acc.authdb /etc/xrootd/Authfile
+if defined STASH_ORIGIN_AUTHFILE
```
Comment (matyasselmeci) [inline configs/config.d/50-stash-origin-authz.cfg]: should these get indented?

Comment (bbockelm) [reply configs/config.d/50-stash-origin-authz.cfg]: yes - can you do an edit for that?

File: configs/config.d/50-stash-origin-authz.cfg [hunk 2]
```diff
@@ -32,12 +32,22 @@ if named stash-origin-auth
 
    sec.protbind * gsi
 
-   acc.authdb /etc/xrootd/Authfile
+   if defined STASH_ORIGIN_AUTHFILE
```
Comment (matyasselmeci) [inline configs/config.d/50-stash-origin-authz.cfg]: Should `STASH_ORIGIN_AUTHFILE` for `stash-origin-auth` here be a different variable than the one used below for `stash-origin`?

File: configs/config.d/50-stash-origin-authz.cfg [hunk 3]
```diff
@@ -32,12 +32,22 @@ if named stash-origin-auth
...
+      acc.authdb $STASH_ORIGIN_AUTHFILE
```
Comment (matyasselmeci) [inline configs/config.d/50-stash-origin-authz.cfg]:
```suggestion
      acc.authdb $STASH_ORIGIN_AUTH_AUTHFILE
```

Comment (brianhlin) [reply configs/config.d/50-stash-origin-authz.cfg]: I understand why but this makes me sad

Comment (bbockelm) [reply configs/config.d/50-stash-origin-authz.cfg]: Can’t stomach the name. How about:

STASH_ORIGIN_PUBLIC_AUTHFILE

below and keep the one here?

**Pull request merged**